daemon/events/filter.go

```diff
@@ -8,19 +8,20 @@ import (
 
 // Filter can filter out docker events from a stream
 type Filter struct {
-	filter    filters.Args
-	getLabels func(id string) map[string]string
+	filter       filters.Args
+	getLabels    func(id string) map[string]string
+	updateFilter func(f *filters.Args)
```

Comment:
I would name this `GetContainerByRef` or something like that (assuming the above change to pass in `dameon.Get()`).